### ISSUE TITLE
Could not update data - Expecting value

### ISSUE THREAD

**Issue by Spankythemusical** [custom-components/grocy#4]
python version -  3.6.8
homeassistant version - 0.95.0
component version - 0.1.1

HA Grocy Integration is throwing the following error, multiple times:

`019-06-28 13:23:20 ERROR (MainThread) [custom_components.grocy] Could not update data - Expecting value: line 1 column 1 (char 0)`

These are the sample responses from the virtual env:

```
>>> from pygrocy import Grocy
>>> grocy = Grocy("http://192.168.x.xxx/api/", "XXXX")
>>> for entry in grocy.stock():
...     print("{} in stock for product id {}".format(entry.product_id, entry.amount))
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
AttributeError: 'Product' object has no attribute 'amount'
>>>
```

```
>>> from pygrocy import Grocy
>>> grocy = Grocy("http://192.168.x.xxx/api/", "XXXX")
>>> for entry in grocy.stock(get_details=True):
...     print("{} in stock for product id {}".format(entry.product_id, entry.amount))
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy.py", line 95, in stock
    item.get_details(self._api_client)
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy.py", line 21, in get_details
    details = api_client.get_product(self.product_id)
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy_api_client.py", line 259, in get_product
    return ProductDetailsResponse(parsed_json)
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy_api_client.py", line 175, in __init__
    self._quantity_unit_purchase = QuantityUnitData(parsed_json.get('quantity_unit_purchase'))
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy_api_client.py", line 13, in __init__
    self._id = parse_int(parsed_json.get('id'))
AttributeError: 'NoneType' object has no attribute 'get'
>>>
```
This is a snippet from the Grocy access.log:

```
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:28:19 +0100] "GET /api/system/db-changed-time HTTP/1.1" 200 234 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:28:19 +0100] "GET /stockoverview HTTP/1.1" 200 12432 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:28:20 +0100] "GET /api/system/db-changed-time HTTP/1.1" 200 233 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:28:20 +0100] "PUT /api/user/settings/auto_night_mode_enabled HTTP/1.1" 204 218 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:28:20 +0100] "GET /api/stock/volatile?expiring_days=5 HTTP/1.1" 200 551 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:28:20 +0100] "GET /api/stock HTTP/1.1" 200 502 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xxx - - [28/Jun/2019:13:28:30 +0100] "GET /stock HTTP/1.1" 404 226 "-" "python-requests/2.22.0"
127.0.1.1:80 192.168.x.xxx - - [28/Jun/2019:13:28:30 +0100] "GET /chores HTTP/1.1" 302 245 "-" "python-requests/2.22.0"
127.0.1.1:80 192.168.x.xxx - - [28/Jun/2019:13:28:30 +0100] "GET /login HTTP/1.1" 200 9967 "-" "python-requests/2.22.0"
127.0.1.1:80 192.168.x.xxx - - [28/Jun/2019:13:29:00 +0100] "GET /stock HTTP/1.1" 404 226 "-" "python-requests/2.22.0"
127.0.1.1:80 192.168.x.xxx - - [28/Jun/2019:13:29:00 +0100] "GET /chores HTTP/1.1" 302 245 "-" "python-requests/2.22.0"
127.0.1.1:80 192.168.x.xxx - - [28/Jun/2019:13:29:00 +0100] "GET /login HTTP/1.1" 200 9967 "-" "python-requests/2.22.0"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:29:20 +0100] "GET /api/system/db-changed-time HTTP/1.1" 200 234 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:29:20 +0100] "GET /stockoverview HTTP/1.1" 200 12432 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:29:22 +0100] "GET /api/system/db-changed-time HTTP/1.1" 200 234 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:29:22 +0100] "PUT /api/user/settings/auto_night_mode_enabled HTTP/1.1" 204 217 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:29:22 +0100] "GET /api/stock/volatile?expiring_days=5 HTTP/1.1" 200 551 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xx - - [28/Jun/2019:13:29:22 +0100] "GET /api/stock HTTP/1.1" 200 502 "http://192.168.x.xxx/stockoverview" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36"
127.0.1.1:80 192.168.x.xxx - - [28/Jun/2019:13:29:31 +0100] "GET /stock HTTP/1.1" 404 226 "-" "python-requests/2.22.0"
127.0.1.1:80 192.168.x.xxx - - [28/Jun/2019:13:29:31 +0100] "GET /chores HTTP/1.1" 302 245 "-" "python-requests/2.22.0"
127.0.1.1:80 192.168.x.xxx - - [28/Jun/2019:13:29:31 +0100] "GET /login HTTP/1.1" 200 9967 "-" "python-requests/2.22.0"
```

And this is a snippet from the Grocy error.log:

```
[Fri Jun 28 13:29:21.644378 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 92, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.645260 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 95, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.645380 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 102, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.645444 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 107, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.645485 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 110, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.646139 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 138, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.646498 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name_plural' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 138, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649153 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 92, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649216 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 95, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649250 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 102, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649273 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 107, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649300 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 110, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649345 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 138, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649360 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name_plural' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 138, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649744 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 92, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649794 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 95, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649827 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 102, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649848 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 107, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649874 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 110, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649921 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 138, referer: http://192.168.x.xxx/stockoverview
[Fri Jun 28 13:29:21.649937 2019] [php7:notice] [pid 134107] [client 192.168.x.xx:51408] PHP Notice:  Trying to get property 'name_plural' of non-object in /var/www/html/data/viewcache/96551edfc0e9ad8293b8482b3fabc97fbc8a6335.php on line 138, referer: http://192.168.x.xxx/stockoverview
```

**Comment by Spankythemusical** [comment]
Just as a gee whiz thing, if I comment out lines 145 and 147 of __init__.py, the error goes away.  If, on the other hand, I comment out 144 and 146, the error is still there.  This leads me to believe that it has something to do with the "chores" section.  Earlier today I deleted every chore and stock item from Grocy, but that did not affect the error.

**Comment by SebRut** [comment]
Could you provide the actual response for stock and chores from your api? You can get them by visiting https://YOUR_GROCY_URL/api#/Stock/get_stock and https://YOUR_GROCY_URL/api#/Chores/get_chores and clicking on "Try it out" and "Execute" at the specified entries. I'd be primarily interested in the content of the response body. Also executing the following code snippet could be interesting:

```python3
from pygrocy import Grocy
grocy = Grocy("http://192.168.x.xxx/api/", "XXXX")

```

**Comment by Spankythemusical** [comment]
Chores:
![image](https://user-images.githubusercontent.com/38551704/60382934-5381d180-9a62-11e9-86b2-fd316375676b.png)
Stock:
![image](https://user-images.githubusercontent.com/38551704/60382942-71e7cd00-9a62-11e9-9a88-7031b1908d05.png)
Code:
```
(homeassistant) homeassistant@UbuntuVM:/home/xxxx$ python
Python 3.6.8 (default, Jan 14 2019, 11:02:34)
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pygrocy import Grocy
>>> grocy = Grocy("http://192.168.x.xxx/api/", "xxxx")
>>> for entry in grocy.stock(get_details=True):
...     print("{} - {}".format(entry.name, entry.available_amount))
...
>>> for entry in grocy.chores(get_details=True):
...     print("{} - {}".format(entry.name, entry.last_tracked_time))
...
>>>
```

**Comment by SebRut** [comment]
Hmmm... do you really don't have products or chores set up? I guess the library should be able this as well. 😅

**Comment by Spankythemusical** [comment]
I didn't have very many set up initially.  I removed everything when the issue started, thinking that maybe it would clear out the database.  I'll throw some junk objects in and run everything again to see what happens.

**Comment by Spankythemusical** [comment]
Chores:
![image](https://user-images.githubusercontent.com/38551704/60383648-82e90c00-9a6b-11e9-8867-e5a2b40350e8.png)

Stock:
![image](https://user-images.githubusercontent.com/38551704/60383653-95634580-9a6b-11e9-8672-4a58c0714d8e.png)

Code:
```
(homeassistant) homeassistant@UbuntuVM:/home/xxxx$ python
Python 3.6.8 (default, Jan 14 2019, 11:02:34)
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pygrocy import Grocy
>>> grocy = Grocy("http://192.168.x.xxx/api/", "xxxx")
>>> for entry in grocy.stock(get_details=True):
...     print("{} - {}".format(entry.name, entry.available_amount))
...
Milk - 1.0
Spam - 2.0
Cereal - 1.0
Toothpaste - 2.0
>>> for entry in grocy.chores(get_details=True):
...     print("{} - {}".format(entry.name, entry.last_tracked_time))
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy.py", line 119, in chores
    chore.get_details(self._api_client)
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy.py", line 58, in get_details
    details = api_client.get_chore(self.chore_id)
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy_api_client.py", line 271, in get_chore
    return ChoreDetailsResponse(parsed_json)
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy_api_client.py", line 212, in __init__
    self._last_done_by = UserDto(parsed_json.get('last_done_by'))
  File "/srv/homeassistant/lib/python3.6/site-packages/pygrocy/grocy_api_client.py", line 79, in __init__
    self._id = parse_int(parsed_json.get('id'))
AttributeError: 'NoneType' object has no attribute 'get'
>>>

```

**Comment by SebRut** [comment]
I think I found the origin of the exception. Parsing chore details for tasks never marked as done fails. https://github.com/SebRut/pygrocy/issues/11

**Comment by SebRut** [comment]
I pushed a new pygrocy and custom component version that hopefully fixes the issue

**Comment by Spankythemusical** [comment]
So the code works:
```
(homeassistant) homeassistant@UbuntuVM:/home/xxx$ python
Python 3.6.8 (default, Jan 14 2019, 11:02:34)
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pygrocy import Grocy
>>> grocy = Grocy("http://192.168.x.xxx/api/", "xxxx")
>>> for entry in grocy.stock(get_details=True):
...     print("{} - {}".format(entry.name, entry.available_amount))
...
Milk - 1.0
Spam - 2.0
Cereal - 1.0
Toothpaste - 2.0
>>> for entry in grocy.chores(get_details=True):
...     print("{} - {}".format(entry.name, entry.last_tracked_time))
...
Dishes - None
Bathroom - None
Clean Baseboards - None
Bins - Regular - None
Bins Recycle - None
>>>
```
But the error remains:
```
019-06-29 14:32:30 ERROR (MainThread) [custom_components.grocy] Could not update data - Expecting value: line 1 column 1 (char 0)
2019-06-29 14:33:01 ERROR (MainThread) [custom_components.grocy] Could not update data - Expecting value: line 1 column 1 (char 0)
2019-06-29 14:33:32 ERROR (MainThread) [custom_components.grocy] Could not update data - Expecting value: line 1 column 1 (char 0)
```

**Comment by eracknaphobia** [comment]
Same issue here. I've added chores (marked them all as completed) and a product. Still getting errors.

**Comment by razzam21** [comment]
> So the code works:
> 
> ```
> (homeassistant) homeassistant@UbuntuVM:/home/xxx$ python
> Python 3.6.8 (default, Jan 14 2019, 11:02:34)
> [GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
> Type "help", "copyright", "credits" or "license" for more information.
> >>> from pygrocy import Grocy
> >>> grocy = Grocy("http://192.168.x.xxx/api/", "xxxx")
> >>> for entry in grocy.stock(get_details=True):
> ...     print("{} - {}".format(entry.name, entry.available_amount))
> ...
> Milk - 1.0
> Spam - 2.0
> Cereal - 1.0
> Toothpaste - 2.0
> >>> for entry in grocy.chores(get_details=True):
> ...     print("{} - {}".format(entry.name, entry.last_tracked_time))
> ...
> Dishes - None
> Bathroom - None
> Clean Baseboards - None
> Bins - Regular - None
> Bins Recycle - None
> >>>
> ```
> 
> But the error remains:
> 
> ```
> 2019-06-29 14:32:30 ERROR (MainThread) [custom_components.grocy] Could not update data - Expecting value: line 1 column 1 (char 0)
> 2019-06-29 14:33:01 ERROR (MainThread) [custom_components.grocy] Could not update data - Expecting value: line 1 column 1 (char 0)
> 2019-06-29 14:33:32 ERROR (MainThread) [custom_components.grocy] Could not update data - Expecting value: line 1 column 1 (char 0)
> ```

I am having the same issue.  I deployed the docker version on a separate machine in an attempt to resolve the issue.  But it persists.

**Comment by razzam21** [comment]
I did some troubleshooting seeing if I could figure something out.  I deployed Grocy in its own container on a separate machine.  Then I modified the access log it was outputting to see if I could discover anything.  I see three requests being made.  To /stock, /chores, and /login.  When I try and go to /stock the log showed a 404 error and the UI did the same thing.  I am not sure what the code does but it looks like /stockoverview may be a better location.  I have filled in the stock into the system.  I hadn't started in on the chores part yet but I plan to in the the future.  I am waiting on a few Amazon Dash Buttons to get here so I can incorporate them into my system.

**Comment by SebRut** [comment]
Thank you for further investigating this. Accessing {GROCYURL}/api/stock should not result in a 404 error. Maybe there is an issue with the docker setup. I checked that endpoint with my own instance and the demo available and both seem to work fine. We could to try posting an issue at the grocy repo to get this to work maybe.

**Comment by razzam21** [comment]
The logs for the container indicate that it is go to {GROCYURL}/stock not {GROCYURL}/api/stock.  When I test {GROCYURL}/api/stock it gives back a json string so that could be the issue that I am experiencing.

**Comment by SebRut** [comment]
Are you maybe missing a slash in the api url in your HA config?

**Comment by razzam21** [comment]
I didn't have a slash in my configuration and now I don't see any errors.  Thanks for your help.

**Comment by SebRut** [comment]
@Spankythemusical could this also be the cause of your problem?

**Comment by Spankythemusical** [comment]
I'll check this evening when I get home and let you know what I find.

**Comment by Spankythemusical** [comment]
hmmm, didn't mean to close this.  

Apologies for taking this long to get around to checking.

```
grocy:
  url: "http://192.168.86.220/api"
  api_key: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
  sensor:
    - enabled: true

```
and 

```
grocy:
  url: "http://192.168.86.220/api/"
  api_key: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
  sensor:
    - enabled: true
```

both result in:

`Could not update data - Expecting value: line 1 column 1 (char 0)`

**Comment by Spankythemusical** [comment]
hmmm, didn't mean to close this.  

Apologies for taking this long to get around to checking.

```
grocy:
  url: "http://192.168.XX.XXX/api"
  api_key: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
  sensor:
    - enabled: true

```
and 

```
grocy:
  url: "http://192.168.XX.XXX/api/"
  api_key: "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
  sensor:
    - enabled: true
```

both result in:

`Could not update data - Expecting value: line 1 column 1 (char 0)`

**Comment by SebRut** [comment]
Hey is the issue still present? We changed some stuff around configuration, could you check out the new instructions please?

**Comment by Spankythemusical** [comment]
It is still present.

```
python --version
Python 3.8.0
```

HACS:
```
Grocy
Installed: v0.3.0
```
configuration.yaml:
```
grocy:
  url: "http://192.168.XX.XXX"
  api_key: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
  verify_ssl: false
  port: 80
  sensor:
    - enabled: true
  binary_sensor:
    - enabled: true
```
Error:
```
grocy: Error on device update!
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.8/site-packages/homeassistant/helpers/entity_platform.py", line 292, in _async_add_entity
    await entity.async_device_update(warning=False)
  File "/srv/homeassistant/lib/python3.8/site-packages/homeassistant/helpers/entity.py", line 448, in async_device_update
    await self.async_update()
  File "/home/homeassistant/.homeassistant/custom_components/grocy/binary_sensor.py", line 33, in async_update
    await self._client.async_update_expiring_products()
  File "/home/homeassistant/.homeassistant/custom_components/grocy/__init__.py", line 198, in async_update_expiring_products
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/srv/homeassistant/lib/python3.8/site-packages/pygrocy/grocy.py", line 161, in expiring_products
    raw_expiring_product = self.volatile_stock().expiring_products
  File "/srv/homeassistant/lib/python3.8/site-packages/pygrocy/grocy.py", line 158, in volatile_stock
    return self._api_client.get_volatile_stock()
  File "/srv/homeassistant/lib/python3.8/site-packages/pygrocy/grocy_api_client.py", line 313, in get_volatile_stock
    parsed_json = resp.json()
  File "/srv/homeassistant/lib/python3.8/site-packages/requests/models.py", line 897, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

http://192.168.XX.XXX/api/stock (truncated):

`[{"product_id":"7","amount":"14","amount_aggregated":"14","best_before_date":"2019-10-06","amount_opened":"0","amount_opened_aggregated":"0","is_aggregated_amount":"0"},{"product_id":"8","amount":"1","amount_aggregated":"1","best_before_date":"2999-12-31","amount_opened":"0","amount_opened_aggregated":"0","is_aggregated_amount":"0"},{"product_id":"9","amount":"1","amount_aggregated":"1","best_before_date":"2999-12-31","amount_opened":"0","amount_opened_aggregated":"0","is_aggregated_amount":"0"},{"product_id":"10","amount":"1","amount_aggregated":"1","best_before_date":"2999-12-31","amount_opened":"0","amount_opened_aggregated":"0","is_aggregated_amount":"0"},{"product_id":"11","amount":"1","amount_aggregated":"1","best_before_date":"2999-12-31","amount_opened":"0","amount_opened_aggregated":"0","is_aggregated_amount":"0"},`

http://192.168.XX.XXX/api/chores (truncated):

`[{"chore_id":"1","last_tracked_time":null,"next_estimated_execution_time":"2019-12-08 10:01:31","track_date_only":"0","next_execution_assigned_to_user_id":null},{"chore_id":"2","last_tracked_time":null,"next_estimated_execution_time":"2019-12-08 10:01:31","track_date_only":"0","next_execution_assigned_to_user_id":null},{"chore_id":"3","last_tracked_time":null,"next_estimated_execution_time":"2019-12-08 10:01:31","track_date_only":"0","next_execution_assigned_to_user_id":null},{"chore_id":"4","last_tracked_time":null,"next_estimated_execution_time":"2019-12-04 10:01:31","track_date_only":"0","next_execution_assigned_to_user_id":null},`

Using previous code:

```
(homeassistant) homeassistant@UbuntuVM:/home/me$ python
Python 3.8.0 (default, Nov  1 2019, 10:26:29)
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pygrocy import Grocy
>>> grocy = Grocy("http://192.168.XX.XXX", "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx", port = 80)
>>>
>>> print("Stock")
Stock
>>> for entry in grocy.stock(get_details=True):
...   print("{} - {}".format(entry.name, entry.available_amount))
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable
>>> print("Chores")
Chores
>>> for entry in grocy.chores(get_details=True):
...   print("{} - {}".format(entry.name, entry.last_tracked_time))
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/srv/homeassistant/lib/python3.8/site-packages/pygrocy/grocy.py", line 199, in chores
    raw_chores = self._api_client.get_chores()
  File "/srv/homeassistant/lib/python3.8/site-packages/pygrocy/grocy_api_client.py", line 327, in get_chores
    parsed_json = resp.json()
  File "/srv/homeassistant/lib/python3.8/site-packages/requests/models.py", line 897, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
>>>

```

**Comment by SebRut** [comment]
_Hey! Just wanted to let you know that I finally accepted, that I'll not be able to maintain this project for the foreseeable future. I'd be happy to hand most of this over to someone willing to work on this integration and will try to support this by working on SebRut/pygrocy._

**Comment by isabellaalstrom** [comment]
Could you please download the latest pre-release and see if this is still present?

**Comment by Spankythemusical** [comment]
Just at a quick glance, it looks as if that issue is resolved.  However, this is now what shows up in the logs:

`2020-08-14 10:49:14 WARNING (MainThread) [homeassistant.util.async_] Detected I/O inside the event loop. This is causing stability issues. Please report issue to the custom component author for grocy doing I/O at custom_components/grocy/config_flow.py, line 89: client.stock()`

**Comment by isabellaalstrom** [comment]
Thanks. Could you add a new issue for the io loop?
